app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/VisibleMessageView.kt

```diff
@@ -243,44 +243,125 @@ class VisibleMessageView : LinearLayout {
+        val lastSentMessageId = mmsSmsDb.getLastSentMessageFromSender(message.threadId, thisUsersSessionId)
+        val isLastSentMessage = lastSentMessageId == message.id
 
-            binding.messageStatusTextView.isVisible = textId != null && (isLastSentMessage || scheduledToDisappear)
-            val showTimer = scheduledToDisappear && !message.isPending
-            binding.messageStatusImageView.isVisible = iconID != null && !showTimer && (!message.isSent || isLastSentMessage)
+        // ----- Case ii.) Message is outgoing but NOT scheduled to disappear -----
+        if (!scheduledToDisappear) {
+            // If this isn't a disappearing message then we never show the timer
+            binding.expirationTimerView.isVisible = false
 
-            binding.messageStatusImageView.bringToFront()
-            binding.expirationTimerView.bringToFront()
-            binding.expirationTimerView.isVisible = showTimer
-            if (showTimer) updateExpirationTimer(message)
-        } else {
-            binding.messageStatusTextView.isVisible = false
-            binding.messageStatusImageView.isVisible = false
+            // If the message has NOT been successfully sent then always show the delivery status text and image..
+            val neitherSentNorRead = !(message.isSent || message.isRead)
+            if (neitherSentNorRead) {
+                binding.messageStatusTextView.isVisible = true
+                binding.messageStatusImageView.isVisible = true
+            }
+            else // ..but if the message HAS been successfully sent or read..
+            {
+                // ..then we'll only display the delivery status text and image if this is the last sent message..
+                if (isLastSentMessage) {
+                    binding.messageStatusTextView.isVisible = true
+                    binding.messageStatusImageView.isVisible = true
+                    binding.messageStatusImageView.bringToFront()
+                }
+                else // ..while hiding it on all other messages.
+                {
+                    binding.messageStatusTextView.isVisible = false
+                    binding.messageStatusImageView.isVisible = false
+                }
+            }
+        }
+        else // ----- Case iii.) Message is outgoing AND scheduled to disappear -----
+        {
+            // If the message is sent or has been read..
+            val sentOrRead = message.isSent || message.isRead
+            if (sentOrRead) {
+                // ..then ALWAYS display the timer for this disappearing message.
+                binding.expirationTimerView.isVisible = true
+                binding.expirationTimerView.bringToFront()
+                updateExpirationTimer(message)
+
+                // And if this is the last sent message then also display the status text (but not the image)..
+                if (isLastSentMessage) {
+                    binding.messageStatusTextView.isVisible = true
```

Comment:
could just remove the if statement entirely.

`binding.messageStatusTextView.isVisible = isLastSentMessage`

Reply:
Nice - done.